### ISSUE TITLE
KIE inference fails w/o error after succesfull training and eval on google colab.

### ISSUE THREAD

**Issue by sandorkonya** [PaddlePaddle/PaddleOCR#8440]
System Environment：[ google colab notebook](https://colab.research.google.com/drive/1ZKKUMsM2NBLV0gySbOUVyOB7f-i4bXuN?usp=sharing)

Paddle：paddlepaddle-gpu==2.3.2  PaddleOCR：paddleocr-2.6.1.1 Related components：see linked notebook

I followed the instructions from the [KIE page](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/kie_en.md) for training.

I can successfully download and extract the XFUND dataset and initiate a training with default setup parameters:

> !python3 tools/train.py -c configs/kie/vi_layoutxlm/ser_vi_layoutxlm_xfund_zh.yml

The checkpoint model gets saved normally:

> /content/PaddleOCR/output/ser_vi_layoutxlm_xfund_zh

The eval script:

> !python3 tools/eval.py -c configs/kie/vi_layoutxlm/ser_vi_layoutxlm_xfund_zh.yml -o Architecture.Backbone.checkpoints=/content/PaddleOCR/output/ser_vi_layoutxlm_xfund_zh/best_accuracy

gives results:

> [2022/11/24 21:40:43] ppocr INFO: metric eval ***************
> [2022/11/24 21:40:43] ppocr INFO: precision:0.7034025895814514
> [2022/11/24 21:40:43] ppocr INFO: recall:0.8414985590778098
> [2022/11/24 21:40:43] ppocr INFO: hmean:0.7662784976217812
> [2022/11/24 21:40:43] ppocr INFO: fps:9.198483312059636

Inference command:

> !python3 tools/infer_kie_token_ser.py -c configs/kie/vi_layoutxlm/ser_vi_layoutxlm_xfund_zh.yml -o Architecture.Backbone.checkpoints=./output/ser_vi_layoutxlm_xfund_zh/best_accuracy Global.infer_img=./ppstructure/docs/kie/input/zh_val_0.jpg

gives no error:

> [2022/11/24 21:48:34] ppocr INFO: Architecture : 
> [2022/11/24 21:48:34] ppocr INFO:     Backbone : 
> [2022/11/24 21:48:34] ppocr INFO:         checkpoints : ./output/ser_vi_layoutxlm_xfund_zh/best_accuracy
> [2022/11/24 21:48:34] ppocr INFO:         mode : vi
> [2022/11/24 21:48:34] ppocr INFO:         name : LayoutXLMForSer
> [2022/11/24 21:48:34] ppocr INFO:         num_classes : 7
> [2022/11/24 21:48:34] ppocr INFO:         pretrained : True
> [2022/11/24 21:48:34] ppocr INFO:     Transform : None
> [2022/11/24 21:48:34] ppocr INFO:     algorithm : LayoutXLM
> [2022/11/24 21:48:34] ppocr INFO:     model_type : kie
> [2022/11/24 21:48:34] ppocr INFO: Eval : 
> [2022/11/24 21:48:34] ppocr INFO:     dataset : 
> [2022/11/24 21:48:34] ppocr INFO:         data_dir : train_data/XFUND/zh_val/image
> [2022/11/24 21:48:34] ppocr INFO:         label_file_list : ['train_data/XFUND/zh_val/val.json']
> [2022/11/24 21:48:34] ppocr INFO:         name : SimpleDataSet
> [2022/11/24 21:48:34] ppocr INFO:         transforms : 
> [2022/11/24 21:48:34] ppocr INFO:             DecodeImage : 
> [2022/11/24 21:48:34] ppocr INFO:                 channel_first : False
> [2022/11/24 21:48:34] ppocr INFO:                 img_mode : RGB
> [2022/11/24 21:48:34] ppocr INFO:             VQATokenLabelEncode : 
> [2022/11/24 21:48:34] ppocr INFO:                 algorithm : LayoutXLM
> [2022/11/24 21:48:34] ppocr INFO:                 class_path : train_data/XFUND/class_list_xfun.txt
> [2022/11/24 21:48:34] ppocr INFO:                 contains_re : False
> [2022/11/24 21:48:34] ppocr INFO:                 order_method : tb-yx
> [2022/11/24 21:48:34] ppocr INFO:                 use_textline_bbox_info : True
> [2022/11/24 21:48:34] ppocr INFO:             VQATokenPad : 
> [2022/11/24 21:48:34] ppocr INFO:                 max_seq_len : 512
> [2022/11/24 21:48:34] ppocr INFO:                 return_attention_mask : True
> [2022/11/24 21:48:34] ppocr INFO:             VQASerTokenChunk : 
> [2022/11/24 21:48:34] ppocr INFO:                 max_seq_len : 512
> [2022/11/24 21:48:34] ppocr INFO:             Resize : 
> [2022/11/24 21:48:34] ppocr INFO:                 size : [224, 224]
> [2022/11/24 21:48:34] ppocr INFO:             NormalizeImage : 
> [2022/11/24 21:48:34] ppocr INFO:                 mean : [123.675, 116.28, 103.53]
> [2022/11/24 21:48:34] ppocr INFO:                 order : hwc
> [2022/11/24 21:48:34] ppocr INFO:                 scale : 1
> [2022/11/24 21:48:34] ppocr INFO:                 std : [58.395, 57.12, 57.375]
> [2022/11/24 21:48:34] ppocr INFO:             ToCHWImage : None
> [2022/11/24 21:48:34] ppocr INFO:             KeepKeys : 
> [2022/11/24 21:48:34] ppocr INFO:                 keep_keys : ['input_ids', 'bbox', 'attention_mask', 'token_type_ids', 'image', 'labels']
> [2022/11/24 21:48:34] ppocr INFO:     loader : 
> [2022/11/24 21:48:34] ppocr INFO:         batch_size_per_card : 2
> [2022/11/24 21:48:34] ppocr INFO:         drop_last : False
> [2022/11/24 21:48:34] ppocr INFO:         num_workers : 0
> [2022/11/24 21:48:34] ppocr INFO:         shuffle : False
> [2022/11/24 21:48:34] ppocr INFO: Global : 
> [2022/11/24 21:48:34] ppocr INFO:     cal_metric_during_train : False
> [2022/11/24 21:48:34] ppocr INFO:     distributed : False
> [2022/11/24 21:48:34] ppocr INFO:     epoch_num : 30
> [2022/11/24 21:48:34] ppocr INFO:     eval_batch_step : [0, 19]
> [2022/11/24 21:48:34] ppocr INFO:     infer_img : ./ppstructure/docs/kie/input/zh_val_0.jpg
> [2022/11/24 21:48:34] ppocr INFO:     kie_det_model_dir : None
> [2022/11/24 21:48:34] ppocr INFO:     kie_rec_model_dir : None
> [2022/11/24 21:48:34] ppocr INFO:     log_smooth_window : 10
> [2022/11/24 21:48:34] ppocr INFO:     print_batch_step : 10
> [2022/11/24 21:48:34] ppocr INFO:     save_epoch_step : 2000
> [2022/11/24 21:48:34] ppocr INFO:     save_inference_dir : None
> [2022/11/24 21:48:34] ppocr INFO:     save_model_dir : ./output/ser_vi_layoutxlm_xfund_zh
> [2022/11/24 21:48:34] ppocr INFO:     save_res_path : /content/output/
> [2022/11/24 21:48:34] ppocr INFO:     seed : 2022
> [2022/11/24 21:48:34] ppocr INFO:     use_gpu : True
> [2022/11/24 21:48:34] ppocr INFO:     use_visualdl : True
> [2022/11/24 21:48:34] ppocr INFO: Loss : 
> [2022/11/24 21:48:34] ppocr INFO:     key : backbone_out
> [2022/11/24 21:48:34] ppocr INFO:     name : VQASerTokenLayoutLMLoss
> [2022/11/24 21:48:34] ppocr INFO:     num_classes : 7
> [2022/11/24 21:48:34] ppocr INFO: Metric : 
> [2022/11/24 21:48:34] ppocr INFO:     main_indicator : hmean
> [2022/11/24 21:48:34] ppocr INFO:     name : VQASerTokenMetric
> [2022/11/24 21:48:34] ppocr INFO: Optimizer : 
> [2022/11/24 21:48:34] ppocr INFO:     beta1 : 0.9
> [2022/11/24 21:48:34] ppocr INFO:     beta2 : 0.999
> [2022/11/24 21:48:34] ppocr INFO:     lr : 
> [2022/11/24 21:48:34] ppocr INFO:         epochs : 30
> [2022/11/24 21:48:34] ppocr INFO:         learning_rate : 5e-05
> [2022/11/24 21:48:34] ppocr INFO:         name : Linear
> [2022/11/24 21:48:34] ppocr INFO:         warmup_epoch : 2
> [2022/11/24 21:48:34] ppocr INFO:     name : AdamW
> [2022/11/24 21:48:34] ppocr INFO:     regularizer : 
> [2022/11/24 21:48:34] ppocr INFO:         factor : 0.0
> [2022/11/24 21:48:34] ppocr INFO:         name : L2
> [2022/11/24 21:48:34] ppocr INFO: PostProcess : 
> [2022/11/24 21:48:34] ppocr INFO:     class_path : train_data/XFUND/class_list_xfun.txt
> [2022/11/24 21:48:34] ppocr INFO:     name : VQASerTokenLayoutLMPostProcess
> [2022/11/24 21:48:34] ppocr INFO: Train : 
> [2022/11/24 21:48:34] ppocr INFO:     dataset : 
> [2022/11/24 21:48:34] ppocr INFO:         data_dir : train_data/XFUND/zh_train/image
> [2022/11/24 21:48:34] ppocr INFO:         label_file_list : ['train_data/XFUND/zh_train/train.json']
> [2022/11/24 21:48:34] ppocr INFO:         name : SimpleDataSet
> [2022/11/24 21:48:34] ppocr INFO:         ratio_list : [1.0]
> [2022/11/24 21:48:34] ppocr INFO:         transforms : 
> [2022/11/24 21:48:34] ppocr INFO:             DecodeImage : 
> [2022/11/24 21:48:34] ppocr INFO:                 channel_first : False
> [2022/11/24 21:48:34] ppocr INFO:                 img_mode : RGB
> [2022/11/24 21:48:34] ppocr INFO:             VQATokenLabelEncode : 
> [2022/11/24 21:48:34] ppocr INFO:                 algorithm : LayoutXLM
> [2022/11/24 21:48:34] ppocr INFO:                 class_path : train_data/XFUND/class_list_xfun.txt
> [2022/11/24 21:48:34] ppocr INFO:                 contains_re : False
> [2022/11/24 21:48:34] ppocr INFO:                 order_method : tb-yx
> [2022/11/24 21:48:34] ppocr INFO:                 use_textline_bbox_info : True
> [2022/11/24 21:48:34] ppocr INFO:             VQATokenPad : 
> [2022/11/24 21:48:34] ppocr INFO:                 max_seq_len : 512
> [2022/11/24 21:48:34] ppocr INFO:                 return_attention_mask : True
> [2022/11/24 21:48:34] ppocr INFO:             VQASerTokenChunk : 
> [2022/11/24 21:48:34] ppocr INFO:                 max_seq_len : 512
> [2022/11/24 21:48:34] ppocr INFO:             Resize : 
> [2022/11/24 21:48:34] ppocr INFO:                 size : [224, 224]
> [2022/11/24 21:48:34] ppocr INFO:             NormalizeImage : 
> [2022/11/24 21:48:34] ppocr INFO:                 mean : [123.675, 116.28, 103.53]
> [2022/11/24 21:48:34] ppocr INFO:                 order : hwc
> [2022/11/24 21:48:34] ppocr INFO:                 scale : 1
> [2022/11/24 21:48:34] ppocr INFO:                 std : [58.395, 57.12, 57.375]
> [2022/11/24 21:48:34] ppocr INFO:             ToCHWImage : None
> [2022/11/24 21:48:34] ppocr INFO:             KeepKeys : 
> [2022/11/24 21:48:34] ppocr INFO:                 keep_keys : ['input_ids', 'bbox', 'attention_mask', 'token_type_ids', 'image', 'labels']
> [2022/11/24 21:48:34] ppocr INFO:     loader : 
> [2022/11/24 21:48:34] ppocr INFO:         batch_size_per_card : 1
> [2022/11/24 21:48:34] ppocr INFO:         drop_last : False
> [2022/11/24 21:48:34] ppocr INFO:         num_workers : 0
> [2022/11/24 21:48:34] ppocr INFO:         shuffle : True
> [2022/11/24 21:48:34] ppocr INFO: profiler_options : None
> [2022/11/24 21:48:34] ppocr INFO: train with paddle 2.3.2 and device Place(gpu:0)
> W1124 21:48:34.903532  1103 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 7.5, Driver API Version: 11.2, Runtime API Version: 10.2
> W1124 21:48:34.910039  1103 gpu_resources.cc:91] device: 0, cuDNN Version: 8.1.
> [2022/11/24 21:48:38] ppocr INFO: resume from ./output/ser_vi_layoutxlm_xfund_zh/best_accuracy
> [2022-11-24 21:48:38,907] [    INFO] - Already cached /root/.paddlenlp/models/layoutxlm-base-uncased/sentencepiece.bpe.model
> [2022-11-24 21:48:39,353] [    INFO] - tokenizer config file saved in /root/.paddlenlp/models/layoutxlm-base-uncased/tokenizer_config.json
> [2022-11-24 21:48:39,353] [    INFO] - Special tokens file saved in /root/.paddlenlp/models/layoutxlm-base-uncased/special_tokens_map.json

I get an empty text file in:

> /content/PaddleOCR/output/ser/xfund_zh/res/infer_results.txt

**Expected behaviour:**
that the inferenced result / output image gets saved in Global.save_res_path.

**Comment by sandorkonya** [comment]
@an1018 , @huangjun12 , @littletomatodonkey any clue?

**Comment by sandorkonya** [comment]
Additional information:

inferencing with the pretrained weights fails similarly with ppstructure:
![image](https://user-images.githubusercontent.com/6852967/203869225-d281b25b-2046-453a-ba49-9a18eafb936c.png)

(i just added a print("iterating over:", image_file) into kie/predict_kie_token_ser.py to see how many images it processes from a folder)

**Comment by sandorkonya** [comment]
Okay. It was the GPU... 

using:
  **--use_gpu False**
arg at the inference solved the problem and i get infernce output.

So simply use:
```
!python3 kie/predict_kie_token_ser.py \
--kie_algorithm=LayoutXLM \
  --ser_model_dir=../inference/ser_vi_layoutxlm_xfund_infer \
  --image_dir=/content/sample_data/img \
  --ser_dict_path=../train_data/XFUND/class_list_xfun.txt \
  --vis_font_path=../doc/fonts/simfang.ttf \
  --ocr_order_method="tb-yx" \
  --rec_model_dir="./ch_PP-OCRv3_rec_infer/" \
  --det_model_dir="./ch_PP-OCRv3_det_infer/" \
  --use_gpu False
```
